mongoengine/queryset/base.py

```diff
@@ -1731,14 +1731,14 @@ def _fields_to_dbfields(self, fields):
                           for x in document._subclasses][1:]
         for field in fields:
             try:
-                field = '.'.join(f.db_field for f in
+                field = '.'.join(f.db_field if not isinstance(f, six.string_types) else f for f in
```

Comment:
Could re-write it without a double-negative, i.e. `f if isinstance(f, six,string_types) else f.db_field`